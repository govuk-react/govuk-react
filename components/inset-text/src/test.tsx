import React from 'react';
import { mount } from 'enzyme';
import Paragraph from '@govuk-react/paragraph';

import { InsetText } from '.';

describe('inset text', () => {
  it('renders provided text', () => {
    const wrapper = mount(
      <InsetText>
        It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.
      </InsetText>
    );

    expect(wrapper.text()).toContain(
      'It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.'
    );
  });

  it('matches the InsetText snapshot', () => {
    expect(
      mount(
        <InsetText>
          It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.
        </InsetText>
      )
    ).toMatchSnapshot('inset text');
  });

  it('can render with a paragraph', () => {
    expect(
      mount(
        <InsetText>
          <Paragraph mb={0}>
            It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the
            application.
          </Paragraph>
        </InsetText>
      )
    ).toMatchSnapshot('inset text with paragraph');
  });
});
