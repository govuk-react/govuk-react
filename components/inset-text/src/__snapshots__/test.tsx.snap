// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inset text can render with a paragraph: inset text with paragraph 1`] = `
.c1 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 0;
}

.c1 > p {
  margin: 0;
}

.c1 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c1 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c1 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding: 15px;
  margin-top: 20px;
  margin-bottom: 20px;
  clear: both;
  border-left: 10px solid #bfc1c3;
}

.c0:first-child {
  margin-top: 0;
}

.c0:only-child,
.c0:last-child {
  margin-bottom: 0;
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-top: 30px;
    margin-bottom: 30px;
  }
}

<InsetTextWithParagraph>
  <InsetText>
    <div
      className="c0"
    >
      <Paragraph
        linkRenderer={[Function]}
        mb={0}
        supportingText={false}
      >
        <src__StyledParagraph
          allowedTypes={
            Array [
              "paragraph",
              "emphasis",
              "strong",
              "link",
              "inlineCode",
              "code",
              "text",
            ]
          }
          escapeHtml={false}
          linkRenderer={[Function]}
          mb={0}
          renderers={
            Object {
              "link": [Function],
            }
          }
          skipHtml={true}
          source="It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application."
          supportingText={false}
        >
          <ReactMarkdown
            allowedTypes={
              Array [
                "paragraph",
                "emphasis",
                "strong",
                "link",
                "inlineCode",
                "code",
                "text",
              ]
            }
            className="c1"
            escapeHtml={false}
            linkRenderer={[Function]}
            mb={0}
            renderers={
              Object {
                "link": [Function],
              }
            }
            skipHtml={true}
            source="It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application."
            supportingText={false}
            transformLinkUri={[Function]}
          >
            <Root
              className="c1"
              key="root-1-1-0"
              node={
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "position": Object {
                            "end": Object {
                              "column": 111,
                              "line": 1,
                              "offset": 110,
                            },
                            "start": Object {
                              "column": 1,
                              "line": 1,
                              "offset": 0,
                            },
                          },
                          "type": "text",
                          "value": "It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.",
                        },
                      ],
                      "position": Object {
                        "end": Object {
                          "column": 111,
                          "line": 1,
                          "offset": 110,
                        },
                        "start": Object {
                          "column": 1,
                          "line": 1,
                          "offset": 0,
                        },
                      },
                      "type": "paragraph",
                    },
                  ],
                  "position": Object {
                    "end": Object {
                      "column": 111,
                      "line": 1,
                      "offset": 110,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 1,
                      "offset": 0,
                    },
                  },
                  "type": "root",
                }
              }
            >
              <div
                className="c1"
              >
                <p
                  key="paragraph-1-1-0"
                >
                  <TextRenderer
                    key="text-1-1-0"
                    node={
                      Object {
                        "position": Object {
                          "end": Object {
                            "column": 111,
                            "line": 1,
                            "offset": 110,
                          },
                          "start": Object {
                            "column": 1,
                            "line": 1,
                            "offset": 0,
                          },
                        },
                        "type": "text",
                        "value": "It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.",
                      }
                    }
                    nodeKey="text-1-1-0"
                    value="It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application."
                  >
                    It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.
                  </TextRenderer>
                </p>
              </div>
            </Root>
          </ReactMarkdown>
        </src__StyledParagraph>
      </Paragraph>
    </div>
  </InsetText>
</InsetTextWithParagraph>
`;

exports[`inset text matches the InsetText snapshot: inset text 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding: 15px;
  margin-top: 20px;
  margin-bottom: 20px;
  clear: both;
  border-left: 10px solid #bfc1c3;
}

.c0:first-child {
  margin-top: 0;
}

.c0:only-child,
.c0:last-child {
  margin-bottom: 0;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-top: 30px;
    margin-bottom: 30px;
  }
}

<InsetTextExample>
  <InsetText>
    <div
      className="c0"
    >
      It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.
    </div>
  </InsetText>
</InsetTextExample>
`;
